clients/src/test/java/org/apache/kafka/clients/producer/KafkaProducerTest.java

```diff
@@ -2397,7 +2399,15 @@ public ProducerRecord<byte[], byte[]> onSend(ProducerRecord<byte[], byte[]> reco
         }
 
         @Override
-        public void onAcknowledgement(RecordMetadata metadata, Exception exception) {
+        public void onAcknowledgement(RecordMetadata metadata, Exception exception, Headers headers) {
+            if (headers == null) {
+                return;
+            }
+            if (!(headers instanceof RecordHeaders)) {
```

Comment:
maybe merge these 2 `If`s instead of having 2 of them

Reply:
sure, fixed at 6c582bd2c5

turn out `headers==null` is not need, it is handled as part of `!(headers instanceof RecordHeaders)`